COLLABORATOR_GUIDE.md

```diff
@@ -73,6 +73,112 @@ All pull requests that modify executable code should be subjected to
+existing code to continue working without modification) are handled as
+semver-minor changes.
+
+Note that errors thrown by dependencies of Node.js (e.g. those originating from
+V8) are generally not under the control of Node.js and therefore are not
+directly subject to this policy. However, care should still be taken when
+landing updates to dependencies when it is known or expected that breaking
+changes to error handling may have been made. Additional CI testing may be
+required.
+
+### When breaking changes actually break things
+
+Breaking changes are difficult primarily because they change the fundamental
+assumptions a user of Node.js has when writing their code and can cause
+existing code to stop functioning as expected -- costing developers and users
+time and energy to fix.
+
+Because breaking (semver-major) changes are permitted to land in master at any
+time, it should be *understood and expected* that at least some subset of the
+user ecosystem *may* be adversely affected *in the short term* when attempting
+to build and use Node.js directly from master. This potential instability is
+precisely why Node.js offers distinct Stable and LTS release streams that carry
+varying explicit stability guarantees. Specifically, as a general rule,
+*breaking* changes should never land in a Stable or LTS branch unless such
+changes are considered to be absolutely necessary to address critical security
+issues.
+
+When any change is landed in master, and it is determined that the such
+changes *do*, in fact, break existing code, a decision may be made to revert
+those changes either temporarily or permanently. However, the decision to
+revert or not can often be based on many complex factors that are not easily
+codified.
+
+## Deprecations
+
+Deprecation refers to the identification of APIs that should no longer be used
+and that may be removed or modified in non-backwards compatible ways in a
+future major release of Node.js.
+
+Node.js uses three fundamental Deprecation levels:
+
+* Soft-Deprecation (or Docs-only deprecation) refers to elements of the Node.js
+  public API that are being staged for deprecation in a future Node.js major
+  release. An explicit notice indicating the deprecated status is added to the
+  API documentation *but no functional changes are implemented in the code*.
+  There will be no runtime deprecation warning.
+
+* Hard-Deprecation refers to the use of warnings being reported at
+  runtime when a deprecated API is used. By default, Node.js will print a
+  warning to stderr the first time a deprecated API is used. A command-line
+  switch can be used to escalate such warnings into runtime errors that will
+  cause the Node.js process to exit. As with Soft-Deprecation, the documentation
+  for the API must be updated to clearly indicate the deprecated status.
+
+* End-of-life refers to APIs that have gone through Hard-Deprecation and are
+  ready to be removed from Node.js entirely.
```

Comment:
We have generally steered clear of put concrete timeline statements in the actual deprecation messages since it's never clear if we'll actually be able to follow through on them.